src/pages/AddItem.js

```diff
@@ -14,24 +34,62 @@ export default function AddItem(props) {
     setPurchaseFrequency(e.target.value);
   };
 
-  const createListItem = (e) => {
-    e.preventDefault();
-    db.collection('generated_token')
-      .add({
-        item_name: itemName,
-        purchase_frequency: parseInt(purchaseFrequency),
-        last_purchased: lastPurchased,
-      })
-      .then((documentReference) => {
-        console.log('document reference ID', documentReference.id);
-      })
-      .catch((error) => {
-        console.log(error.message);
-      });
-    setItemName('');
-    setLastPurchased(null);
+  const checkForDuplicates = () => {
+    // we would like to potentially normalize white space but unsuccessfull so far
+    const normalizedUserInput = itemName
+      .toLowerCase()
+      .replace(/[^\w\s]|_/g, '');
+
+    const matchingItemName = listItems.filter((item) => {
+      const normalizedDataBaseItem = item.item_name
+        .toLowerCase()
+        .replace(/[^\w\s]|_/g, '');
+      console.log(normalizedDataBaseItem);
+      return normalizedDataBaseItem === normalizedUserInput;
+    });
+
+    if (matchingItemName.length > 0) {
+      return false;
+    } else {
+      return true;
+    }
   };
 
+  async function createListItem(e) {
+    e.preventDefault();
+    try {
+      const result = await checkForDuplicates(itemName);
+      if (result) {
+        db.collection('generated_token')
+          .add({
+            item_name: itemName,
+            purchase_frequency: parseInt(purchaseFrequency),
+            last_purchased: lastPurchased,
+          })
+          .then((documentReference) => {
+            console.log('document reference ID', documentReference.id);
+          })
```

Comment:
Is this serving a purpose now that the list is functioning? Perhaps we could remove this, which would help make it less busy.

Reply:
I'm not sure exactly why `.then((documentReference))`  is there; was it a debugging tool when the other team was setting up the Firestore in the first week? Either way, we can definitely remove it. I'm all for getting rid of code we don't need!

Reply:
We got rid of 
```javascript 
.then((documentReference) => {
    console.log('document reference ID', documentReference.id);
})
```

And no werid side effects or anything!